### PR TITLE
init UploadStream Header

### PR DESCRIPTION
**Pull request by Jazz86**

### CODE REVIEW

File: src/Header/UploadStream.h [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+#include <string>
+
+// write file to local may use a lock file or system call to signify that a certain file is currently in use.
+class UploadStream
+{
+public:
+    UploadStream();
+    // filename will include file path, if run successfully, it will return 1, else 0;
+    int writeFile(std::string filename, std::string &binary);
```
Comment (john0312) [inline src/Header/UploadStream.h]: const ref

File: src/Header/UploadStream.h [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+#ifndef GROUP4_HEADER_UploadStream_H_
+#define GROUP4_HEADER_UploadStream_H_
+
+#include <string>
+
+// write file to local may use a lock file or system call to signify that a certain file is currently in use.
+class UploadStream
+{
+public:
+    UploadStream();
```
Comment (john0312) [inline src/Header/UploadStream.h]: copyable?

File: src/Header/UploadStream.h (hunk 1 above)
Comment (john0312) [reply src/Header/UploadStream.h]: Please comment that this is appending to the file. Consider calling it `appendToFile`
Consider putting the filename in constructor if this class maps to a file.

File: src/Header/UploadStream.h [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    // filename will include file path, if run successfully, it will return 1, else 0;
+    int writeFile(std::string filename, std::string &binary);
+};
```
Comment (john0312) [inline src/Header/UploadStream.h:30]: Add fd or FILE* for the file?

File: src/Header/UploadStream.h (hunk 1 above)
Comment (Jazz86) [reply src/Header/UploadStream.h]: done.

File: src/Header/UploadStream.h (hunk 2 above)
Comment (Jazz86) [reply src/Header/UploadStream.h]: non copyable is implemented.

File: src/Header/UploadStream.h (hunk 3 above)
Comment (Jazz86) [reply src/Header/UploadStream.h:30]: fd is added. done.

File: src/Header/UploadStream.h [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+  // user should handle the order yourself. If it run successfully, it will
+  // return 1 else 0.
+  int appendToFile(const std::string& binary);
+  // non copyable
+  UploadStream(const UploadStream&) = delete;
+  UploadStream& operator=(const UploadStream&) = delete;
+
+ private:
+  std::fstream fd;
+  std::string filename;
```
Comment (john0312) [inline src/Header/UploadStream.h:29]: Please add variable comment.

File: src/Header/UploadStream.h [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+
+#include <fstream>
+#include <string>
+
+// write file to local may use a lock file or system call to signify that a
+// certain file is currently in use.
+class UploadStream {
+ public:
+  // filename will include file path,
+  UploadStream(std::string filename);
```
Comment (john0312) [inline src/Header/UploadStream.h]: const ref?

File: src/Header/UploadStream.h (hunk 4 above)
Comment (Jazz86) [reply src/Header/UploadStream.h:29]: done.

File: src/Header/UploadStream.h (hunk 5 above)
Comment (Jazz86) [reply src/Header/UploadStream.h]: done.

**Review by john0312 — APPROVED** [review]
Approved, you are free to merge this PR.

**Pull request merged**